tests/test_project.py

```diff
@@ -871,6 +871,106 @@ def test_update_submodules_strategy(repos_tmpdir):
     assert net_tools_project.sha('HEAD', cwd=kconfiglib_dst_dir) \
            == kconfiglib_new_sha
 
+@pytest.mark.xfail
+def test_update_submodules_relpath(tmpdir):
+    # Regression test for
+    # https://github.com/zephyrproject-rtos/west/issues/545.
+    #
+    # We need to make sure that submodules with relative paths are
+    # initialized correctly even when ther is no "origin" remote.
+    #
+    # Background: unless the config variable git.$SUBMODULENAME.url is
+    # set, git relies on the "default remote" for figuring out how to
+    # turn that relative path into something useful. The "default
+    # remote" ends up being "origin" in the situation where we are
+    # initializing a submodule from a superproject which is a west
+    # project. However, west neither makes nor expects any guarantees
+    # about any particular git remotes being present in a project, so
+    # that only works if we set the config variable before
+    # initializing the module.
+
+    # The following paths begin with 'project' if they are west projects,
+    # and 'submodule' if they are direct or transitive submodules of a
+    # west project. Numeric suffixes reflect the tree structure of the
+    # workspace:
+    #
+    # workspace
+    # ├── manifest
+    # ├── project-1
+    # │   └── submodule-1-1
+    # └── project-2
+    #     ├── submodule-2-1
+    #     └── submodule-2-2
+    #         └── submodule-2-2-1
+    remotes = tmpdir / 'remotes'  # parent directory for 'remote' repos
+    remote_repositories = [
+        remotes / path for path in
+        ['manifest',
+         'project-1',
+         'submodule-1-1',
+         'project-2',
+         'submodule-2-1',
+         'submodule-2-2',
+         'submodule-2-2-1',
```

Comment:
I'm trying to think of different names for "top" submodules known to west versus sub-submodules not know to west. This is obviously a big difference. How about `top-submodule-x-y` and `rec-submodule-x-y-z`? Or `project-1-submodule-1` for the top ones?

Reply:
I think the names are clear enough as-is, especially given the generous comments.

Reply:
`submodule-2-2` and `submodule-2-2-1` look the same. They look like one is "only one level down" and that's it. But from a west perspective they are totally different. The last one is _invisible_ to west.

Reply:
I am sorry, but I find this comment very nitpicky. This is just a test case, and a heavily commented test case at that. It isn't like this is user-facing documentation. Can we please just leave this be?